plugins/flatpak/gs-plugin-flatpak.c

```diff
@@ -461,13 +461,108 @@ _build_transaction (GsPlugin *plugin, GsFlatpak *flatpak,
+{
+	FlatpakInstallation *installation;
+	g_autoptr (GFile) installation_dir = NULL;
+	g_autoptr (GFileInfo) info = NULL;
+
+	installation = gs_flatpak_get_installation (flatpak);
+	installation_dir = flatpak_installation_get_path (installation);
+
+	info = g_file_query_filesystem_info (installation_dir,
+					     G_FILE_ATTRIBUTE_FILESYSTEM_FREE,
+					     NULL,
+					     error);
+	if (info == NULL)
+		return FALSE;
+
+	*free_space = g_file_info_get_attribute_uint64 (info, G_FILE_ATTRIBUTE_FILESYSTEM_FREE);
+	return TRUE;
+}
+
+static gboolean
+gs_flatpak_has_space_to_install (GsFlatpak *flatpak, GsApp *app)
+{
+	g_autoptr(GError) error = NULL;
+	guint64 free_space = 0;
+	guint64 min_free_space = 0;
+	guint64 space_required = 0;
+	FlatpakInstallation *installation;
+
+	space_required = gs_app_get_size_download (app);
+	if (space_required == GS_APP_SIZE_UNKNOWABLE) {
+		g_warning ("Failed to query download size: %s", gs_app_get_unique_id (app));
+		space_required = 0;
+	}
+
+	installation = gs_flatpak_get_installation (flatpak);
+
+	if (!flatpak_installation_get_min_free_space_bytes (installation, &min_free_space, &error)) {
+		g_autoptr(GFile) installation_file = flatpak_installation_get_path (installation);
+		g_autofree gchar *path = g_file_get_path (installation_file);
+		g_warning ("Error getting min-free-space config value of OSTree repo at %s:%s", path, error->message);
+		g_clear_error (&error);
+	}
+	space_required = space_required + min_free_space;
+
+	if (!get_installation_dir_free_space (flatpak, &free_space, &error)) {
+		g_warning ("Error getting the free space available for installing %s: %s",
+			   gs_app_get_unique_id (app), error->message);
+		/* Even if we fail to get free space, we don't want to block this user-intiated
+		 * install action. It might happen that there is enough space to install but
+		 * an error happened during querying the filesystem info. */
+		return TRUE;
+	}
+
+	return free_space >= space_required;
+}
+
+static gboolean
+gs_flatpak_has_space_to_update (GsFlatpak *flatpak, GsAppList *list, gboolean is_auto_update)
+{
+	g_autoptr(GError) error = NULL;
+	guint64 free_space = 0;
+	guint64 min_free_space = 0;
+	guint64 space_required = 0;
+	FlatpakInstallation *installation;
+
+	if (is_auto_update) {
+		for (guint i = 0; i < gs_app_list_length (list); i++) {
+			GsApp *app_temp = gs_app_list_index (list, i);
+			space_required += gs_app_get_size_installed (app_temp);
+		}
+	}
+
+	installation = gs_flatpak_get_installation (flatpak);
+
+	if (!flatpak_installation_get_min_free_space_bytes (installation, &min_free_space, &error)) {
+		g_autoptr(GFile) installation_file = flatpak_installation_get_path (installation);
+		g_autofree gchar *path = g_file_get_path (installation_file);
+		g_warning ("Error getting min-free-space config value of OSTree repo at %s:%s", path, error->message);
+		g_clear_error (&error);
+	}
+	space_required = (space_required * 2) + min_free_space;
```

Comment:
As per https://github.com/endlessm/gnome-software/commit/cf0a5f18c0a2ce22250ba1340f3b732436fd871d , 
we can drop `* 2` factor, or are you planning to squash that change after-wards?

Reply:
I’ll merge that in later (currently juggling branches).